schema.cc

```diff
@@ -1709,6 +1722,156 @@ bytes token_column_computation::compute_value(const schema& schema, const partit
     return {long_type->decompose(long_value)};
 }
 
+bytes collection_column_computation::serialize() const {
+    rjson::value serialized = rjson::empty_object();
+    const char* type = nullptr;
+    switch (_kind) {
+        case kind::keys:
+            type = "collection_keys";
+            break;
+        case kind::values:
+            type = "collection_values";
+            break;
+        case kind::entries:
+            type = "collection_entries";
+            break;
+    }
+    rjson::add(serialized, "type", rjson::from_string(type));
+    rjson::add(serialized, "collection_name", rjson::from_string(sstring(_collection_name.begin(), _collection_name.end())));
+    return to_bytes(rjson::print(serialized));
+}
+
+column_computation_ptr collection_column_computation::for_target_type(std::string_view type, const bytes& collection_name) {
+    if (type == "collection_keys") {
+        return collection_column_computation::for_keys(collection_name).clone();
```

Comment:
I don't understand what this clone() is about, but I vaguely remember asking this somewhere else so maybe I have an answer waiting elsewhere :-)

Reply:
Clone method for `column_computation` assures that we can "copy" a unique pointer held as a pointer to base class (`std::unique_ptr<column_computation>`) even though what we have is in fact a pointer to a derived class. So, `clone` was already needed, and here I use it conveniently.